x/evm/keeper/state_transition.go

```diff
@@ -242,29 +224,21 @@ func (k *Keeper) ApplyMessage(evm *vm.EVM, msg core.Message, cfg *params.ChainCo
 		return nil, stacktrace.Propagate(sdkerrors.Wrap(types.ErrVMExecution, vmErr.Error()), "vm execution failed")
 	}
 
+	gasUsed := msg.Gas() - leftoverGas
 	return &types.MsgEthereumTxResponse{
 		Ret:      ret,
 		Reverted: false,
+		GasUsed:  gasUsed,
 	}, nil
 }
 
-// CheckGasConsumption verifies that the amount of gas consumed so far matches the intrinsic gas value.
-func (k *Keeper) CheckGasConsumption(msg core.Message, cfg *params.ChainConfig, gasConsumed uint64, isContractCreation bool) error {
+// GetEthIntrinsicGas get the transaction intrinsic gas cost
+func (k *Keeper) GetEthIntrinsicGas(msg core.Message, cfg *params.ChainConfig, isContractCreation bool) (uint64, error) {
 	height := big.NewInt(k.ctx.BlockHeight())
 	homestead := cfg.IsHomestead(height)
 	istanbul := cfg.IsIstanbul(height)
 
-	intrinsicGas, err := core.IntrinsicGas(msg.Data(), msg.AccessList(), isContractCreation, homestead, istanbul)
-	if err != nil {
-		// should have already been checked on Ante Handler
-		return stacktrace.Propagate(err, "intrinsic gas failed")
-	}
-
-	if intrinsicGas != gasConsumed {
-		return sdkerrors.Wrapf(types.ErrInconsistentGas, "expected gas consumption to be %d (intrinsic gas only), got %d", intrinsicGas, gasConsumed)
-	}
```

Comment:
can we add a test to check this logic?

Reply:
ok I will see what I can do. There is currently no test file for the EVM. Since it just call a function from geth, should I hardcode the correct value?

Reply:
let's create an issue to add a test for this logic

Reply:
yes, created there
https://github.com/tharsis/ethermint/issues/200